src/Ocelot/Configuration/SecurityOptions.cs

```diff
@@ -4,31 +4,31 @@ public class SecurityOptions
     {
         public SecurityOptions()
         {
-            IPAllowedList = new();
-            IPBlockedList = new();
+            IPAllowedList = new List<string>();
+            IPBlockedList = new List<string>();
         }
 
         public SecurityOptions(string allowed = null, string blocked = null)
             : this()
         {
             if (!string.IsNullOrEmpty(allowed))
             {
-                IPAllowedList.Add(allowed);
+                IPAllowedList = IPAllowedList.Append(allowed).ToList();
```

Comment:
??? Could you explain?

Reply:
@ggnaegi The issue lies in the design of the [IPSecurityPolicy](https://github.com/ThreeMammals/Ocelot/pull/2170/files#diff-d964726ca860b79e9aaf467165b3ecc847a810d3a8c6f0c81a4200f334a32a11), which relies on processing string objects, often referred to as a flawed design (aka Chinese design). It is my firm belief that IP policy processing should be based on `IpAddress` objects instead of solely relying on strings.

> So what's the relation with the 1 million of strings in memory?

This is a problem of the design issue mentioned above, descendant problem. The [Creator class logic](https://github.com/ThreeMammals/Ocelot/pull/2170/files#diff-7ce752e07b4e6102e72adac03d23a9bbfaf87bf18320ae962e7258d78c9354a3R26-R27) parses the `IpAddress object and converts it into strings.
Let's consider the following test cases:
- the range `192.168.1.1-192.168255.255`: it produces 256<sup>2</sup>, a maximum of 65 536 strings in the list
- the range `10.1.1.1-10.255.255.255`: it generates 256<sup>3</sup>, a maximum of 16 777 216 string objects in the list, resulting in 16 million items❗ This highlights a significant problem.

However, this design issue is out of the current scope. Let the author to complete this config-feat.

Reply:
@Fabman08 Do you understand this code review issue?
Would you prefer to keep calling base constructor `this()` but remove producing new list `.ToList()`?
Or would you remove `this()` but wanna keep `.ToList()`? 😃

Reply:
@raman-m Yes, I understand the issue reported, and I choose to keep calling `this()` and remove `.ToList()`.
Fixed in last commit

Reply:
I will add [my concerns](https://github.com/ThreeMammals/Ocelot/pull/2170#discussion_r1848536987) as a new task for the next refactoring round.